misc-scripts/xref_mapping/XrefParser/HPAParser.pm

```diff
@@ -39,70 +42,61 @@ sub run {
   my $source_id    = $ref_arg->{source_id};
   my $species_id   = $ref_arg->{species_id};
   my $files        = $ref_arg->{files};
-  my $verbose      = $ref_arg->{verbose};
-  my $dbi          = $ref_arg->{dbi};
-  $dbi = $self->dbi unless defined $dbi;
-
-  if((!defined $source_id) or (!defined $species_id) or (!defined $files) ){
-    croak "Need to pass source_id, species_id, files and rel_file as pairs";
-  }
-  $verbose |=0;
-
-  my $filename = @{$files}[0];
+  my $verbose      = $ref_arg->{verbose} || 0;
+  my $dbi          = $ref_arg->{dbi} || $self->dbi;
 
-  my $file_io = $self->get_filehandle($filename);
-  if ( !defined($file_io) ) {
-    return 1;
-  }
-
-  my $parsed_count = 0;
+  croak "Need to pass source_id, species_id, files and rel_file as pairs"
+    unless defined $source_id and defined $species_id and defined $files;
 
-  $file_io->getline();
+  my $file = @{$files}[0];
 
-  while ( defined( my $line = $file_io->getline() ) ) {
+  my $file_io = $self->get_filehandle($file);
+  croak "Could not open $file\n" unless defined $file_io;
 
-    $line =~ s/\s*$//;
-		
+  my $input_file = Text::CSV->new({ sep_char           => ",",
+				    empty_is_undef     => 1
+				  }) or croak "Cannot use file $file: ".Text::CSV->error_diag ();
+  
+  $file_io->getline(); # skip header # Antibody,antibody_id,ensembl_peptide_id,link
 
-    my ( $antibody, $antibody_id, $ensembl_peptide_id, $link) = split( /,/, $line );
-
-    if ( !defined($antibody) || !defined($ensembl_peptide_id) ) {
-      printf( "Line %d contains  has less than two columns.\n",
-	      1 + $parsed_count );
-      print ("The parsing failed\n");
-      return 1;
-    }
+  my $parsed_count = 0;
+  while ( my $row = $input_file->getline( $file_io ) ) {
+    my ( $antibody, $antibody_id, $ensembl_peptide_id, $link );
+    ( $antibody = $row->[0] ) =~ s/\s*$//;
+    ( $antibody_id = $row->[1] ) =~ s/\s*$//;
+    ( $ensembl_peptide_id = $row->[2] ) =~ s/\s*$//;
+    ( $link = $row->[3] ) =~ s/\s*$//;
+
+    croak sprintf("Line %d contains has less than two columns.\nParsing failed", 1 + $parsed_count)
+      unless defined $antibody and defined $ensembl_peptide_id;
 	
-    my $label       = $antibody;
-    my $type        = 'translation';
-    my $description = '';
-    my $version     = '1';
-
-    ++$parsed_count;
+    my $label   = $antibody;
+    my $type    = 'translation';
+    my $version = '1';
 
     my $xref_id = $self->get_xref( $antibody_id, $source_id, $species_id, $dbi );
 
-    if ( !defined($xref_id) || $xref_id eq '' ) {
-      $xref_id = $self->add_xref({ acc        => $antibody_id,
-				   version    => $version,
-				   label      => $label,
-				   desc       => $description,
-				   source_id  => $source_id,
-				   species_id => $species_id,
-                                   dbi        => $dbi,
-				   info_type  => "DIRECT"} );
+    unless(defined $xref_id or $xref_id ne '') {
```

Comment:
Leaving aside the fact that I am really not a fan of using unless for anything more complicated than "unless defined $foo", especially with double negatives (the "ne" bit), I am quite sure this condition is in fact different than what it was before the change - (NOT a) OR (NOT b) <=> NOT (a **AND** b). That would imply either the original version was wrong or your current one is. Could you double-check, please?

Oh, and one way or another I would suggest using q{} to indicate an empty string. Or just replace the "string defined + non-empty" condition with a truthiness check, I am pretty sure MySQL does not like 0 in auto-increment numeric columns so we needn't worry about the string "0".